### PR TITLE
fix constrained MALIS logic to change affinities before computing wei…

### PR DESCRIPTION
**Pull request by srinituraga**
fix constrained MALIS logic to change affinities before computing weights, then use the weights correctly. The weights need to be multiplied separately with positive and negative elementary loss functions. I wonder if we should update the `malis_loss_op` to use `tf.losses.mean_squared_error`, which actually takes a `weight` parameter.

https://www.tensorflow.org/api_docs/python/tf/losses/mean_squared_error

### CODE REVIEW

File: mala/losses/malis_loss.py [hunk 1]
```diff
@@ -17,10 +17,14 @@ def get_edge_weights(self, affs, gt_affs, gt_seg):
...
         weights_pos = self.malis_pass(affs, gt_affs, gt_seg, pos=1)
 
-        return weights_neg + weights_pos
+        return (weights_neg, weights_pos)
 
     def malis_pass(self, affs, gt_affs, gt_seg, pos):
 
+        # in the positive pass (pos==1), we set boundary edges (gt_aff==0) to their true values
+        # in the negative pass (pos==0), we set non-boundary edges (gt_aff==1) to their true values
+        affs[gt_affs == (1-pos)] = 1-pos
```
Comment (funkey) [inline mala/losses/malis_loss.py:26]: This unintentionally changes the passed predictions (passed as reference, not copy).

**Comment by funkey** [comment]
As discussed elsewhere, most of the logic is actually fine. The only thing missing is setting the predictions to the correct values for the positive and negative pass, fixed in 2e0fb0cad85978c03a2c59aaec52b9194b583125.
